codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordDecoder.java

```diff
@@ -90,6 +94,57 @@ protected DnsRecord decodeRecord(
             String name, DnsRecordType type, int dnsClass, long timeToLive,
             ByteBuf in, int offset, int length) throws Exception {
 
+        if (type == DnsRecordType.A) {
```

Comment:
`DnsRecordType.A` is a `static final DnsRecordType`, seems can't be use as a case statement label 

```java
public class DnsRecordType implements Comparable<DnsRecordType> {

    /**
     * Address record RFC 1035 Returns a 32-bit IPv4 address, most commonly used
     * to map hostnames to an IP address of the host, but also used for DNSBLs,
     * storing subnet masks in RFC 1101, etc.
     */
    public static final DnsRecordType A = new DnsRecordType(0x0001, "A");
```